src/Memory.cs

```diff
@@ -106,11 +157,58 @@ public unsafe Span<byte> GetSpan()
         ///
         /// Therefore, the returned span should not be used after calling the grow method or
         /// after calling into WebAssembly code.
+        /// 
+        /// Note that WebAssembly always uses little endian as byte order. On platforms 
+        /// that use big endian, you will need to convert numeric values accordingly.
         /// </remarks>
         public unsafe Span<T> GetSpan<T>(int address)
             where T : unmanaged
         {
-            return MemoryMarshal.Cast<byte, T>(GetSpan()[address..]);
+            return GetSpan<T>(address, checked((int)((GetLength() - address) / sizeof(T))));
+        }
+
+        /// <summary>
+        /// Gets a span of a section of the memory.
+        /// </summary>
+        /// <returns>Returns the span of a section of the memory.</returns>
+        /// <param name="address">The zero-based address of the start of the span.</param>
+        /// <param name="length">The length of the span.</param>
+        /// <remarks>
+        /// The span may become invalid if the memory grows.
+        ///
+        /// This may happen if the memory is explicitly requested to grow or
+        /// grows as a result of WebAssembly execution.
+        ///
+        /// Therefore, the returned span should not be used after calling the grow method or
+        /// after calling into WebAssembly code.
+        /// 
+        /// Note that WebAssembly always uses little endian as byte order. On platforms 
+        /// that use big endian, you will need to convert numeric values accordingly.
+        /// </remarks>
+        public unsafe Span<T> GetSpan<T>(long address, int length)
+            where T : unmanaged
+        {
+            if (address < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(address));
+            }
+
+            if (length < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(length));
+            }
+
+            var context = store.Context;
+            var data = Native.wasmtime_memory_data(context.handle, this.memory);
+            var memoryLength = this.GetLength();
+
+            // Note: A Span<T> can span more than 2 GiB bytes if sizeof(T) > 1.
+            long byteLength = (long)length * sizeof(T);
+
+            if (address > memoryLength - byteLength)
```

Comment:
Formatting nit: could you put braces around the if body?

Reply:
Fixed.